src/Arcus.Security.Providers.AzureKeyVault/KeyVaultSecretProvider.cs

```diff
@@ -167,43 +166,100 @@ public virtual async Task<Secret> GetSecretAsync(string secretName)
 
+            Task<Secret> getSecretTask;
             if (_isUsingAzureSdk)
             {
-                Secret secret = await GetSecretUsingSecretClientAsync(secretName);
-                return secret;
+                getSecretTask = GetSecretUsingSecretClientAsync(secretName);
             }
             else
             {
-                Secret secret = await GetSecretUsingKeyVaultClientAsync(secretName);
-                return secret;
+                getSecretTask = GetSecretUsingKeyVaultClientAsync(secretName);
             }
+
+            Logger.LogTrace("Getting a secret {SecretName} from Azure Key Vault {VaultUri}...", secretName, VaultUri);
+            var secret = await getSecretTask;
+            Logger.LogTrace("Got secret from Azure Key Vault {VaultUri}", VaultUri);
+
+            return secret;
+        }
+
+        /// <summary>
+        /// Stores a secret value with a given secret name
+        /// </summary>
+        /// <param name="secretName">The name of the secret</param>
+        /// <param name="secretValue">The value of the secret</param>
+        /// <returns>Returns a <see cref="Secret"/> that contains the latest information for the given secret</returns>
+        /// <exception cref="ArgumentException">The <paramref name="secretName"/> must not be empty</exception>
+        /// <exception cref="ArgumentNullException">The <paramref name="secretName"/> must not be null</exception>
+        /// <exception cref="ArgumentException">The <paramref name="secretValue"/> must not be empty</exception>
+        /// <exception cref="ArgumentNullException">The <paramref name="secretValue"/> must not be null</exception>
+        /// <exception cref="SecretNotFoundException">The secret was not found, using the given name</exception>
+        /// <exception cref="KeyVaultErrorException">The call for a secret resulted in an invalid response</exception>
+        public virtual async Task<Secret> StoreSecretAsync(string secretName, string secretValue)
+        {
+            Guard.NotNullOrWhitespace(secretName, nameof(secretName), "Requires a non-blank secret name to request a secret in Azure Key Vault");
+            Guard.NotNullOrWhitespace(secretValue, nameof(secretValue), "Requires a non-blank secret value to store a secret in Azure Key Vault");
+            Guard.For<FormatException>(() => !SecretNameRegex.IsMatch(secretName), "Requires a secret name in the correct format to request a secret in Azure Key Vault, see https://docs.microsoft.com/en-us/azure/key-vault/general/about-keys-secrets-certificates#objects-identifiers-and-versioning");
+
+            Task<Secret> storeSecretTask;
+            if (_isUsingAzureSdk)
+            {
+                storeSecretTask = SetSecretUsingSecretClientAsync(secretName, secretValue);
+            }
+            else
+            {
+                storeSecretTask = StoreSecretUsingKeyVaultClientAsync(secretName, secretValue);
```

Comment:
Misalignment of the method-names: **Set**SecretUsing... vs **Store**SecretUsing...
Any specific reason for this?

Reply:
Yes, sloppyness 😅